File: app/src/main/java/com/main/climbingdiary/view/chart/XAxisValueFormatter.java
package com.main.climbingdiary.view.chart;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;

import java.util.ArrayList;

public class XAxisValueFormatter implements IAxisValueFormatter {

    private ArrayList<String> mValues;

    XAxisValueFormatter(ArrayList<String> values) {
        this.mValues = values;
    }

    @Override
    public String getFormattedValue(float value, AxisBase axis) {
        // "value" represents the position of the label on the axis (x or y)
        try {
            return mValues.get((int) value);
        }catch(Exception e){
            return "False";
        }
    }
}

File: app/src/main/java/com/main/climbingdiary/chart/XAxisValueFormatter.kt
package com.main.climbingdiary.chart

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.ValueFormatter
import java.util.*

class XAxisValueFormatter(private val values: ArrayList<String>) : ValueFormatter() {
    
    override fun getFormattedValue(value: Float): String {
        return value.toString()
    }

    override fun getAxisLabel(value: Float, axis: AxisBase): String {
        if (value.toInt() >= 0 && value.toInt() <= values.size - 1) {
            return values[value.toInt()]
        } else {
            return ("").toString()
        }
    }
}